analyses/cell-type-ewings/template_notebooks/singler-workflow/01-singler-results.Rmd

```diff
@@ -0,0 +1,335 @@
+---
+title: "Summary of tumor cell classification with `SingleR` for `r params$library_id`"
+author: Ally Hawkins
+date: "`r Sys.Date()`"
+output:
+  html_document:
+    toc: true
+    toc_depth: 3
+    code_folding: "hide"
+params:
+  library_id: NULL
+  sce_file: NULL
+  singler_results_file: NULL
+  geneset_scores_file: NULL
+  marker_genes_file: NULL
+---
+
+This report summarizes the classification of `r params$library_id` using the following approach: 
+
+- Tumor cells were identified by using `AUCell`. 
+These cells are labeled as `tumor_original`. 
+- All other cells were labeled using `SingleR` with a combination of three references: 
+  - All tumor cells annotated by `AUCell` for all other libraries in `SCPCP000015`. 
+  Any cells that are labeled as tumor using this reference are labeled as `tumor_new`. 
+  - `BlueprintEncodeData` from `celldex`
+  - `HumanPrimaryCellAtlasData` from `celldex`
+
+
+## Setup
+
+```{r}
+# check that sce and results files exist 
+stopifnot(
+  "sce file does not exist" = file.exists(params$sce_file),
+  "SingleR results file does not exist" = file.exists(params$singler_results_file), 
+  "Gene set scores file does not exist" = file.exists(params$geneset_scores_file),
+  "Marker genes file does not exist" = file.exists(params$marker_genes_file)
+)
+```
+
+
+```{r packages}
+suppressPackageStartupMessages({
+  # load required packages
+  library(SingleCellExperiment)
+  library(ggplot2)
+})
+
+# Set default ggplot theme
+theme_set(
+  theme_bw()
+)
+
+# quiet messages
+options(readr.show_col_types = FALSE)
+ComplexHeatmap::ht_opt(message = FALSE)
+```
+
+
+```{r base paths}
+# The path to this module
+module_base <- rprojroot::find_root(rprojroot::is_renv_project)
+
+# source in helper functions: plot_gene_heatmap() and plot_cnv_heatmap() 
+# create_classification_df() and create_marker_gene_df()
+validation_functions <- file.path(module_base, "scripts", "utils", "tumor-validation-helpers.R")
+source(validation_functions)
+```
+
+```{r}
+# Read in files
+sce <- readr::read_rds(params$sce_file)
+singler_results_df <- readr::read_tsv(params$singler_results_file)
+geneset_scores_df <- readr::read_tsv(params$geneset_scores_file)
+```
+
+## Prep for plots
+
+```{r}
+# original library annotation 
+orig_tumor <- glue::glue("tumor-{params$library_id}")
+
+# generate classification df to use for plots 
+classification_df <- sce |> 
+  scuttle::makePerCellDF(use.dimred = "UMAP") |> 
+  # replace UMAP.1 with UMAP1
+  dplyr::rename_with(
+    \(x) stringr::str_replace(x, "^UMAP\\.", "UMAP")
+  ) |>
+  # get rid of excess columns
+  dplyr::select(barcodes, UMAP1, UMAP2) |> 
```

Comment:
I don't know why I haven't suggested this before, but maybe could to simplify this to:
```suggestion
  # get rid of excess columns and rename UMAPs
  dplyr::select(barcodes, UMAP1 = UMAP.1, UMAP2 = UMAP.2) |> 
```